### ISSUE TITLE
Investigate why `daemon off;` and `master_process off;` are required for nginx to work with the module

### ISSUE THREAD

**Issue by nponsard** [Nitrokey/nethsm-pkcs11#93]

**Comment by nponsard** [comment]
`softhsm` works perfectly without these options, it looks like a problem with the tokio runtime. I tried the blocking and the non blocking versions, both crashes.

**Comment by nponsard** [comment]
https://github.com/tokio-rs/tokio/issues/4301

**Comment by ansiwen** [comment]
> This seems to confirm my guess : [tokio-rs/tokio#4301](https://github.com/tokio-rs/tokio/issues/4301)
> 
> Forking when a tokio runtime is runnning is not supported.

That leads to the question: why is the module initialized by nginx _before_ the daemonize fork? Would be better to postpone it. Maybe there is already a possibility in nginx or openssl config to do the module initialization later?

**Comment by nponsard** [comment]
> That leads to the question: why is the module initialized by nginx _before_ the daemonize fork? Would be better to postpone it. Maybe there is already a possibility in nginx or openssl config to do the module initialization later?

It's because Nginx does a test on the configuration before starting, the test includes checking that the private key (on the NetHSM) and the certificate exist and match.

**Comment by nponsard** [comment]
Look into how openssl is initialized ?

**Comment by nponsard** [comment]
Check if nginx has an option to "reload" OpenSSL

**Comment by nponsard** [comment]
maybe an option to not check the configuration when starting ?

**Comment by nponsard** [comment]
The difference is that both functions that work only does a lock on the internal DB whereas C_Sign does an HTTP request
